crates/matrix-sdk-indexeddb/src/crypto_store/mod.rs

```diff
@@ -1273,6 +1273,12 @@ impl_crypto_store! {
             }
         }
     }
+
+    async fn clear_caches(&self) {
+        self.session_cache.clear()
+        // We don't need to clear static_account as it only contains immutable data
```

Comment:
```suggestion
        // We don't need to clear `static_account` as it only contains immutable data
```

Reply:
no, the static account only contains immutable data about the account. The parts of the account changing are in the store cache, that's correctly reset upon regeneration of the olm machine.